src/TLuaInterpreter.cpp

```diff
@@ -6629,42 +6630,113 @@ int TLuaInterpreter::getCharacterName(lua_State* L)
 // Documentation: https://wiki.mudlet.org/w/Manual:Miscellaneous_Functions#getProfileInformation
 int TLuaInterpreter::getProfileInformation(lua_State* L)
 {
+    QString info;
     Host& host = getHostFromLua(L);
-    QString info = host.readProfileData(qsl("description"));
+    const int params = lua_gettop(L);
+
+    switch (params) {
+        case 0:
+        {
+            info = host.readProfileData(qsl("description"));
+            break;
+        }
+        case 1:
+        {
+            QString profileName = getVerifiedString(L, __func__, 1, "profile name");
+            if (profileName.isEmpty()) {
+                lua_pushnil(L);
+                lua_pushstring(L, "getProfileInformation: profile name cannot be empty");
+                return 2;
+            }
+            if (!mudlet::self()->profileExists(profileName)) {
+                lua_pushnil(L);
+                lua_pushfstring(L, "getProfileInformation: profile '%s' does not exist", profileName.toUtf8().constData());
+                return 2;
+            } else {
+                info = mudlet::self()->readProfileData(profileName, qsl("description"));
+            }
+            break;
+        }
+        default:
+        {
+            lua_pushfstring(L, "getProfileInformation: Too many arguments");
+            lua_pushboolean(L, false);
+            return 2;
+        }
```

Comment:
Generally speaking we do NOT fail on too MANY arguments because we have often used extra arguments when we discover in the future that we need to tweak/twiddle/bloody frob things to accommodate something we didn't think about when we first created the Lua API function.